### PR TITLE
Blast hsp panel tracks and interactions

### PR DESCRIPTION
**Pull request by its-mahdi**
- [x] What is the link to the Jira that this PR is for?
https://www.ebi.ac.uk/panda/jira/browse/TRM-24479
- [x] Have you written tests and do these have >= 75% coverage of the code that you are contributing?
- [x] Is it possible to visually evaluate this PR in the browser? If so, what are the steps to do this (eg specific URLs or sections)?
Look at the HSP details panel.
- [ ] Are there any linting errors that required you to add exceptions?
No
- [ ] If there are backend requests, have you checked for the existence of the resource's attributes before attempting access?
No

### CODE REVIEW

File: src/tools/blast/components/results/BlastResult.tsx [hunk 1]
```diff
@@ -151,6 +157,12 @@ const BlastResult = () => {
 
   const [selectedEntries, setSelectedEntries] = useState<string[]>([]);
   const [urlParams, setUrlParams] = useState<URLResultParams>();
+  const [hspDetail, setHspDetail] = useState<BlastHsp>();
```
Comment (xwatkins) [inline src/tools/blast/components/results/BlastResult.tsx:160]: I think `hspDetail`, `hitLength` and `hspDetailPanel` could be consolidated into one hook

File: src/tools/blast/components/results/HSPDetailPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+    hsp_query_to,
+    hsp_qseq,
+    hsp_hit_from,
+    hsp_hit_to,
+    hsp_hseq,
+    hsp_identity,
+  } = hsp;
+  // TODO calculate actual length based on total match and query lengths
+  // const actualLength = hsp_align_len;
+  const actualLength = hitLength;
```
Comment (xwatkins) [inline src/tools/blast/components/results/HSPDetailPanel.tsx:60]: As hitLength is a prop, this is now superfluous, just use `hitLength`

File: src/tools/blast/components/results/HSPDetailPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+            color: '#014371',
+            opacity,
+          },
+        ];
+
+        // eslint-disable-next-line no-param-reassign
+        node.data = [...lineSegments, ...blockSegments];
+      }
+    },
+    [hsp_query_from, hsp_query_to, hsp_identity, hitLength]
```
Comment (xwatkins) [inline src/tools/blast/components/results/HSPDetailPanel.tsx:137]: Missing variables in the array of values that the effect depends on - see eslint output to know which ones.

File: src/tools/blast/components/results/HSPDetailPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+          name: 'Query',
+          sequence: hsp_qseq,
+        },
+        {
+          name: 'Match',
+          sequence: hsp_hseq,
+        },
+      ];
+
+      node.onActiveTrackChange = (trackId: string) => {
```
Comment (xwatkins) [inline src/tools/blast/components/results/HSPDetailPanel.tsx:195]: I don't think we need this for BLAST, only for Align

File: src/tools/blast/components/results/HSPDetailPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+  }
+
+  if (loading) {
+    return <Loader />;
+  }
+
+  return (
+    <SlidingPanel position="bottom">
+      <>
+        {/* TODO move to stylesheet */}
```
Comment (xwatkins) [inline src/tools/blast/components/results/HSPDetailPanel.tsx:261]: Indeed

File: src/tools/blast/components/results/BlastResult.tsx (hunk 1 above)
Comment (its-mahdi) [reply src/tools/blast/components/results/BlastResult.tsx:160]: Done

File: src/tools/blast/components/results/HSPDetailPanel.tsx (hunk 2 above)
Comment (its-mahdi) [reply src/tools/blast/components/results/HSPDetailPanel.tsx:61]: Done

File: src/tools/blast/components/results/HSPDetailPanel.tsx (hunk 3 above)
Comment (its-mahdi) [reply src/tools/blast/components/results/HSPDetailPanel.tsx:106]: Done

File: src/tools/blast/components/results/HSPDetailPanel.tsx (hunk 4 above)
Comment (its-mahdi) [reply src/tools/blast/components/results/HSPDetailPanel.tsx:165]: Removed

File: src/tools/blast/components/results/HSPDetailPanel.tsx (hunk 5 above)
Comment (its-mahdi) [reply src/tools/blast/components/results/HSPDetailPanel.tsx]: Moved these styles to its own file for HSP details panel, but let me know if they should have gone to `SlidingPanel` styles.

**Pull request merged**